src/game/spawn.rs

```diff
@@ -0,0 +1,835 @@
+//! Initialization of level content entities
+
+use super::{
+	animation::*, components::*, drawing::*, inputs::CycleInteraction, level::*,
+	logic::ComputedCycleTurnability, prelude::*,
+};
+use crate::{
+	assets::*,
+	graphics::*,
+	send_event,
+	ui::hover::{self, *},
+	AppSet,
+};
+use bevy::{
+	math::bounding::{Aabb2d, BoundingCircle},
+	sprite::Anchor,
+};
+use rand::Rng as _;
+use std::f32::consts::{PI, TAU};
+
+pub(super) fn plugin(app: &mut App) {
+	use LevelInitializationSet::*;
+	app.init_resource::<LastLevelSessionId>()
+		.init_resource::<ExpiringLevelSessionId>()
+		.add_event::<EnterLevel>()
+		.add_event::<SpawnLevel>()
+		.configure_sets(
+			Update,
+			(
+				ReceiveCommand,
+				(SpawnPrimaryEntities, AfterReceiveCommand).after(ReceiveCommand),
+				(SpawnVisualEntities, AfterSpawnPrimaryEntities).after(SpawnPrimaryEntities),
+				AfterSpawnVisualEntities.after(SpawnVisualEntities),
+			)
+				.after(AppSet::ExecuteInput)
+				.before(AppSet::GameLogic),
+		)
+		.add_systems(
+			Update,
+			(
+				handle_enter_level.in_set(ReceiveCommand),
+				despawn_expired_level_entities
+					.run_if(resource_changed::<ExpiringLevelSessionId>)
+					.in_set(AfterReceiveCommand),
+				(spawn_primary_level_entities, spawn_thing_entities)
+					.chain()
+					.in_set(SpawnPrimaryEntities),
+				(
+					set_vertex_transforms,
+					set_thing_transforms,
+					set_cycle_transforms,
+					init_thing_animation,
+					init_cycle_hover_hints,
+					init_thing_hover_hints,
+				)
+					.in_set(AfterSpawnPrimaryEntities),
+				(
+					create_vertex_visuals,
+					create_cycle_visuals,
+					create_link_visuals,
+					create_thing_sprites,
+					create_box_color_markers,
+					create_button_color_markers,
+				)
+					.in_set(SpawnVisualEntities),
+				init_cycle_animation.in_set(AfterSpawnVisualEntities),
+				(
+					apply_level_hint_text,
+					(
+						|mut history: ResMut<MoveHistory>| history.clear(),
+						|mut is_completed: ResMut<IsLevelCompleted>| is_completed.0 = false,
+						|mut completion: ResMut<LevelCompletionConditions>| *completion = default(),
+						send_event(GameLayoutChanged),
+					)
+						.run_if(on_event::<SpawnLevel>()),
+				)
+					.in_set(AfterReceiveCommand),
+			),
```

Comment:
A) This is starting to look like a Schedule, that we should run in an exclusive system with 
`world.run_schedule(level_setup_schedule)`
B) ~~I think you may need to put in some apply_system_buffers to flush your spawn commands and such~~ seems that bevy does this automatically based on .after() .before() and .chain()

Reply:
I thought about the schedule thing as well, and I liked it for our current needs, but it would cost us the convenience of quick initialization of newly spawned entities (for example, when a black box spits out a new entity, we would want systems like `create_thing_sprites` running on every update). Though I suppose that could still be deferred to another schedule

Reply:
Sounds good.